cloud/spawn.go

```diff
@@ -69,23 +69,13 @@ func (so *SpawnOptions) validate(settings *evergreen.Settings) error {
 	}
 
 	// validate public key
-	rsa := "ssh-rsa"
-	dss := "ssh-dss"
-	isRSA := strings.HasPrefix(so.PublicKey, rsa)
-	isDSS := strings.HasPrefix(so.PublicKey, dss)
-	if !isRSA && !isDSS {
-		return errors.Errorf("Invalid spawn options: "+
-			"either an invalid Evergreen-managed key name has been provided,"+
-			"or the key value does not start with %s or %s", rsa, dss)
+	if err = evergreen.ValidateSSHKey(so.PublicKey); err != nil {
+		return errors.Wrap(err, "Invalid spawn options: ")
```

Comment:
`errors.Wrap` already formats error strings with colons between each message, so you only need to put `"invalid spawn options"` as the wrapping context string and it'll look like `invalid spawn options: <wrapped error message>`.